### DISCUSSION TITLE
Timer fires only once

### DISCUSSION THREAD

**Discussion by s8ngyu** [ANLAB-KAIST/KENSv3#229] [Project Q&A 2. Transport Layer (TCP)] [answered]
When SYNACK being sent in packetArrived, I added a timer for it waiting for ACK.
![CleanShot 2024-05-07 at 06 51 38@2x](https://github.com/ANLAB-KAIST/KENSv3/assets/41620322/e251c8d7-eb50-4137-83a0-04ed14dfa68b)

As you can see in this packet log, after sending SYNACK for the first time, module sends ACK right away but it has checksum corruption. After 100000000 nano seconds, it sends SYNACK again but it has checksum corruption.
![CleanShot 2024-05-07 at 07 01 45@2x](https://github.com/ANLAB-KAIST/KENSv3/assets/41620322/523bd2c2-e2bc-455d-a10b-f0c299f3d5eb)

Since it never received ACK for SYNACK, it needs to send SYNACK one more time after 100000000 nano seconds (that's what solution do) but timer never gets called even I never canceled it.

**Comment by Pusnow** [comment, marked as answer]
> Since it never received ACK for SYNACK, it needs to send SYNACK one more time after 100000000 nano seconds (that's what solution do) but timer never gets called even I never canceled it.

Once a timer is fired, you need to reregister a new timer. The timers are not periodic.
